packages/wasm-terminal/lib/process/process.ts

```diff
@@ -34,30 +34,26 @@ export default class Process {
     }
 
     if (commandOptions.module) {
-      this.wasiCommand = new WASICommand(
+      this.command = new WASICommand(
         commandOptions,
         sharedStdin,
         startStdinReadCallback
       );
     } else {
-      this.callbackCommand = commandOptions.callback;
+      this.command = new CallbackCommand(commandOptions);
     }
   }
 
   async start(pipedStdinData?: Uint8Array) {
-    if (this.wasiCommand) {
+    if (this.command instanceof WASICommand) {
```

Comment:
This should be abstracted by the `Command` class.

The process should not know about if what was provided was a `WASICommand` or a `CallbackCcommand`. Both should work the same way through the same interface.